assets/sip-big_vector/big_vector.move

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Typus Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module typus_framework::big_vector {
+    use std::vector;
+
+    use sui::object::{Self, UID};
+    use sui::tx_context::TxContext;
+    use sui::dynamic_field;
+
+    // ======== Errors ========
+
+    const E_NOT_EMPTY: u64 = 0;
+
+    // ======== Structs ========
+
+    struct BigVector<phantom Element> has key, store {
+        id: UID,
+        slice_count: u64,
+        slice_size: u64,
+        length: u64,
+    }
+
+    // ======== Functions ========
+
+    /// create BigVector
+    public fun new<Element: store>(slice_size: u64, ctx: &mut TxContext): BigVector<Element> {
+        let id = object::new(ctx);
+        let slice_count = 1;
+        dynamic_field::add(&mut id, slice_count, vector::empty<Element>());
+        BigVector<Element> {
+            id,
+            slice_count,
+            slice_size,
+            length: 0,
+        }
+    }
+
+    /// return the slice_count of the BigVector
+    public fun slice_count<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.slice_count
+    }
+
+    /// return the max size of a slice in the BigVector
+    public fun slice_size<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.slice_size
+    }
+
+    /// return the size of the BigVector
+    public fun length<Element: store>(bv: &BigVector<Element>): u64 {
+        bv.length
+    }
+
+    /// return the slice_id related to the index i
+    public fun slice_id<Element: store>(bv: &BigVector<Element>, i: u64): u64 {
+        (i / bv.slice_size) + 1
```

Comment:
The 1-indexing on slices is a little strange to me, is there a reason for that, vs 0-indexing?

Reply:
There was a purpose in the past but no longer exists, it should be 0-indexing now